File: java-source/src/main/java/com/example/client/ExampleClientRPC.java
package com.example.client;

import com.example.state.IOUState;
import com.google.common.net.HostAndPort;
import kotlin.Pair;
import net.corda.client.rpc.CordaRPCClient;
import net.corda.core.messaging.CordaRPCOps;
import net.corda.core.transactions.SignedTransaction;
import org.apache.activemq.artemis.api.core.ActiveMQException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import rx.Observable;

import java.util.List;
import java.util.concurrent.ExecutionException;

/**
 * Demonstration of using the CordaRPCClient to connect to a Corda Node and
 * steam some State data from the node.
 */
public class ExampleClientRPC {
    public static void main(String[] args) throws ActiveMQException, InterruptedException, ExecutionException {
        if (args.length != 1) {
            throw new IllegalArgumentException("Usage: ExampleClientRPC <node address>");
        }

        final Logger logger = LoggerFactory.getLogger(ExampleClientRPC.class);
        final HostAndPort nodeAddress = HostAndPort.fromString(args[0]);
        final CordaRPCClient client = new CordaRPCClient(nodeAddress, null, null);

        // Can be amended in the com.example.Main file.
        client.start("user1", "test");
        final CordaRPCOps proxy = client.proxy();

        // Grab all signed transactions and all future signed transactions.
        final Pair<List<SignedTransaction>, Observable<SignedTransaction>> txsAndFutureTxs =
                proxy.verifiedTransactions();
        final List<SignedTransaction> txs = txsAndFutureTxs.getFirst();
        final Observable<SignedTransaction> futureTxs = txsAndFutureTxs.getSecond();

        // Log the 'placed' IOUs and listen for new ones.
        futureTxs.startWith(txs).toBlocking().subscribe(
                transaction ->
                        transaction.getTx().getOutputs().forEach(
                                output -> {
                                    final IOUState iouState = (IOUState) output.getData();
                                    logger.info(iouState.getIOU().toString());
                                })
        );
    }
}

File: python-source/src/main/python/com/example/client/ExampleClientRPC.py
import sys
from net.corda.client.rpc import CordaRPCClient
from net.corda.core.contracts import ContractState
from net.corda.core.utilities import NetworkHostAndPort
from org.slf4j import LoggerFactory

class ExampleClientRPC:
    logger = LoggerFactory.getLogger(__name__)

    def __init__(self):
        pass

    def log_state(self, state):
        self.logger.info("{}", state.state.data)

    def main(self):
        if len(sys.argv) != 2:
            raise TypeError("USAGE: ./jython.sh src/main/python/com/example/client/ExampleClientRPC.py [HOST:ARTEMIS_PORT]")

        node_address = NetworkHostAndPort.parse(sys.argv[1])
        client = CordaRPCClient(node_address)
        proxy = client.start("user1", "test").getProxy()

        snapshot, updates = proxy.vaultTrack(ContractState)

        print(snapshot)

if __name__ == '__main__':
    ExampleClientRPC().main()